### ISSUE TITLE
determine_reindex_operator_against_reference: identity operation when tested gives misleading results

### ISSUE THREAD

**Issue by graeme-winter** [dials/dials#2451]
Tested this out in a different context, and part of this was testing data against itself: the `a,b,c` operator did not score well which surprises me greatly

```
===> a,b,c
a,b,c 8900 0.6949341199682955
-b,a+b+c,-a 275 0.9838242855595567
y-z,-x+y,y 276 0.9838246694543971
a+b+c,-a,-b 30 0.9991592758815085
x-y,x-z,x 68 0.7644956156811127
y,y-z,-x+y 78 0.7566363676126439
x-z,x,x-y 51 0.9846857051045993
a,c,-a-b-c 76 0.7656359571156953
a,-a-b-c,b 33 0.988277541252158
c,a,b 57 0.9874080517395264
b,c,a 104 0.7569313255039162
b,-a-b-c,c 311 0.9927016904651287
-a-b-c,a,c 311 0.9927016904651302
c,b,-a-b-c 50 0.7654592957603776
-a-b-c,b,a 48 0.7680467368918933
-a-b-c,c,b 45 0.6794968003788755
x,x-y,x-z 4771 0.3966515925737752
c,-a-b-c,a 243 0.0065125163867377775
-c,-b,-a 28 -0.00021824109549658833
b,a,-a-b-c 292 0.012545320022837213
-a,-b,a+b+c 61 0.16885836283737685
-a,-c,-b 61 0.5090859361585583
-x+y,y,y-z 316 0.9533101731278606
-b,-a,-c 326 0.9559541329377925
===> a+b+c,-a,-b
```

It would seem obvious to me that the identify should give the best CC between data sets, the fact that it does not feels like a problem <investigating>

**Comment by graeme-winter** [comment]
OK, there _is_ something weird here

The underlying implementation is 

```
    cc = flex.linear_correlation(s0.data(), s1.data())
    print(cc.coefficient(), cc.n())
```

which gives 1.0 and NN as expected 🤔

**Comment by graeme-winter** [comment]
```
    x, y = s0.common_sets(other=s1)
    cc = flex.linear_correlation(x.data(), y.data())
    print(cc.coefficient(), cc.n())
```

is sufficient to reproduce the problem now; implementation of `common_sets`

**Comment by graeme-winter** [comment]
TL;DR if you give `common_sets` two copies of the same thing it has a different code path to the same object twice; the matches are comprehensively different. Now trying to reduce this to a compact bug report.

**Comment by jbeilstenedmands** [comment]
@graeme-winter Is this a dials issue - seems that we don't use the function in this way in the code? Perhaps in 
`determine_reindex_operator_against_reference` we can have a quick check if the two objects are the same, and if so do a deepcopy?
